terminal-tui/src/components.rs

```diff
@@ -215,3 +218,242 @@ impl MockComponent for Shortcuts {
+                    Attribute::Custom("divider"),
+                    AttrValue::String(String::new()),
+                )
+                .unwrap_string();
+            let spans = match self.props.get(Attribute::Text).map(|x| x.unwrap_payload()) {
+                Some(PropPayload::Vec(spans)) => spans
+                    .iter()
+                    .cloned()
+                    .map(|x| x.unwrap_text_span())
+                    .map(|x| {
+                        // Keep colors and modifiers, or use default
+                        let (fg, bg, modifiers) = utils::use_or_default_styles(&self.props, &x);
+                        Spans::from(vec![TuiSpan::styled(
+                            x.content,
+                            Style::default().add_modifier(modifiers).fg(fg).bg(bg),
+                        )])
+                    })
+                    .collect(),
+                _ => Vec::new(),
+            };
+            let selected = self.state().unwrap_one().unwrap_usize();
+
+            let tabs = TuiTabs::new(spans)
+                .style(Style::default().fg(foreground))
+                .highlight_style(Style::default().fg(highlighted))
+                .divider(divider)
+                .select(selected);
+            render.render_widget(tabs, area);
+        }
+    }
+
+    fn query(&self, attr: Attribute) -> Option<AttrValue> {
+        self.props.get(attr)
+    }
+
+    fn attr(&mut self, attr: Attribute, value: AttrValue) {
+        self.props.set(attr, value)
+    }
+
+    fn state(&self) -> State {
+        State::One(StateValue::Usize(self.states.index))
+    }
+
+    fn perform(&mut self, cmd: Cmd) -> CmdResult {
+        match cmd {
+            Cmd::Move(Direction::Right) => {
+                let prev = self.states.index;
+                self.states.incr_tab_index(true);
+                if prev != self.states.index {
+                    CmdResult::Changed(self.state())
+                } else {
+                    CmdResult::None
+                }
+            }
+            _ => CmdResult::None,
+        }
+    }
+}
+
+/// A container with tabs and a component for each tab.
+pub struct TabContainer<Id>
+where
+    Id: Eq + Hash,
+{
+    pub props: Props,
+    pub tabs: Tabs,
+    pub children: Vec<(String, Id, Box<dyn MockComponent>)>,
+}
+
+impl<Id> Default for TabContainer<Id>
+where
+    Id: Eq + Hash,
+{
+    fn default() -> Self {
+        Self {
+            props: Props::default(),
+            tabs: Tabs::default(),
+            children: vec![],
+        }
+    }
+}
+
+impl<Id> TabContainer<Id>
+where
+    Id: Eq + Hash,
+{
+    pub fn children(mut self, children: Vec<(String, Id, Box<dyn MockComponent>)>) -> Self {
+        let titles = children
+            .iter()
+            .map(|(title, _, _)| TextSpan::from(title.clone()))
+            .collect::<Vec<_>>();
+        self.children = children;
+        self.tabs = Tabs::default().spans(&titles);
+        self
+    }
+}
```

Comment:
`tabbed_ui.child("comments", CommentComponent::new())`

Reply:
Took this into account in the rework.